src/txn/txn_rollback_to_stable.c

```diff
@@ -91,7 +93,23 @@ __rollback_abort_update(WT_SESSION_IMPL *session, WT_ITEM *key, WT_UPDATE *first
         if (upd->txnid == WT_TXN_ABORTED)
             continue;
 
-        if (rollback_timestamp < upd->durable_ts || upd->prepare_state == WT_PREPARE_INPROGRESS) {
+        /*
+         * The unstable update according to the following needs to be aborted.
+         * 1. An invisible update according to the checkpoint snapshot during recovery.
+         * 2. The update durable timestamp is greater than the stable timestamp.
+         * 3. A prepared update.
+         *
+         * In a live system, there are no concurrent transaction running along with
+         * rollback to stable to verify transaction id of an update to confirm the stability
+         * of an update. But during recovery, the transaction id of an update must be validated
+         * against the checkpoint snapshot to confirm whether it is stable.
+         *
+         * Usually during recovery, there are no in memory updates present on the page. But
+         * whenever an unstable fast truncate operation is written to the disk, as part
+         * of the rollback to stable page read, it instantiates the tombstones on the page.
+         */
```

Comment:
The comment talks about a live system versus during recovery. Does the conditional need to check that we're in recovery or not? If this function knows/assumes it is in recovery, I would suggest stating that in the comment and removing the comparison to the live system. If it can be either live or recovery, then do we need a check here?

Reply:
The function internally checks and avoids performing transaction id checks other than recovery. I added a comment to mention that the transaction id check is ignored in all scenarios other than recovery and removed the live system comment.